### PR TITLE
Display welcome message on all pages

### PR DESCRIPTION
**Pull request by hannako**
This PR relates to: https://trello.com/c/wyFRwDcF/17-message-on-homepage-to-explain-change-to-users

- Shows a single blue banner on all pages containing a link to provide feedback and an explanation about the new dgu service. Once dismissed, the standard beta banner is displayed.

Bug fixes:
- Styling on static pages
- Reinstates beta banner on all pages (was missing from landing and static pages)

### CODE REVIEW

File: app/assets/stylesheets/components.scss [hunk 1]
```diff
@@ -34,6 +34,76 @@
...
+  }
+  a:hover {
+    font-weight: bold;
+  }
+}
+
+// when beta message is displayed, spacing of adjacent elements must be adjusted____________
+
+// 1. Static pages and home page
+.dgu-beta-message__show {
```
Comment (maxf) [inline app/assets/stylesheets/components.scss]: you can group everything under the top level rule on line 38, using &__show {...}

Comment (maxf) [inline app/assets/stylesheets/components.scss]: According to BEM naming conventions (http://getbem.com/naming/) the name should be
`.dgu-beta-message--show` as `show` is a modifier for the class

File: app/assets/stylesheets/components.scss [hunk 2]
```diff
@@ -34,6 +34,76 @@
...
+      }
+    }
+  }
+}
+
+// 3. Search results page
+.dgu-beta-message__show {
+  form {
+    .dgu-split-background__top {
+      .dgu-split-background__top-inner {
```
Comment (maxf) [inline app/assets/stylesheets/components.scss]: You can use `&__top-inner` to avoid duplication

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -11,4 +12,13 @@ def set_raven_context
                         environment: Rails.env,
                         app: ENV['VCAP_APPLICATION'])
   end
+
+  def toggle_beta_message
+    flash[:beta_message] =
+      beta_message_unseen? ? 'show' : 'hide'
```
Comment (maxf) [inline app/controllers/application_controller.rb]: it seems that 'hide' is not needed as a value and could be empty

**Comment by TobyRet** [comment]
I'll make the changes in Jess's absence

File: app/controllers/application_controller.rb (hunk 3 above)
Comment (TobyRet) [reply app/controllers/application_controller.rb]: flash[:beta_message] needs to be assigned something regardless of whether beta_message_unseen? return true or false

File: app/controllers/search_controller.rb [hunk 4]
```diff
@@ -1,5 +1,6 @@
 class SearchController < LoggedAreaController
   before_action :search_for_dataset, only: [:search]
+  before_action :toggle_beta_message
```
Comment (laurentqro) [inline app/controllers/search_controller.rb]: already present in `ApplicationController` and `LoggedAreaController` inherits from it, so this `before_action` is not need I think.

**Comment by andrewgarner** [comment]
beta

Is the bold styling required when you hover over the links? It's a bit jarring having it wrap each time.

**Comment by TobyRet** [comment]
Yes I thought that should start on a new line @hannako ?

**Comment by andrewgarner** [comment]
Even with a line break we still might see the same behaviour depending on device width.

**Comment by maxf** [comment]
I'll check with the front-end community. That's appeared before.

**Comment by maxf** [comment]
@hannako . Alternating bold / normal is an anti-pattern because of this problem, but also because it makes some users (specifically users of assistive tech like screen magnification) confused because things change on the screen unexpectedly and with no specific reason.

File: app/controllers/search_controller.rb (hunk 4 above)
Comment (hannako) [reply app/controllers/search_controller.rb]: 👍

**Comment by andrewgarner** [comment]
There is a path through this that doesn't quite work when the application is in private beta.

You are shown the message on the "How we’ll use your data" which you see if you haven't opted into the beta. The "get in touch" link is private to the beta, so clicking on it just takes you back to the page you're already on. There is no way to get in touch without joining the beta.

**Pull request merged**